read_tag_block.py

```diff
@@ -1,35 +1,85 @@
 from pycomm.ab_comm.clx import Driver as ClxDriver
 from nio.block.base import Block, Signal
-from nio.block.mixins.enrich.enrich_signals import EnrichSignals
+from nio.block.mixins import EnrichSignals, Retry
 from nio.properties import VersionProperty, StringProperty, Property
 
 
-class ReadTag(EnrichSignals, Block):
+class ReadTag(EnrichSignals, Retry, Block):
 
-    version = VersionProperty("0.1.4")
-    host = StringProperty(title='Host Address')
-    tags = Property(title='Tags')
+    host = StringProperty(title='Host Address', order=0)
+    tags = Property(title='Tags', order=1)
+    version = VersionProperty('0.2.0')
 
     def __init__(self):
         super().__init__()
-        self.cnxn = ClxDriver()
+        self.cnxn = None
+
+    def before_retry(self, *args, **kwargs):
+        self._disconnect()
+        self._connect()
 
     def configure(self, context):
         super().configure(context)
-        self.cnxn.open(self.host())
+        try:
+            self._connect()
+        except Exception:
+            self.cnxn = None
+            msg = 'Unable to connect to {}'.format(self.host())
+            self.logger.exception(msg)
 
     def stop(self):
         super().stop()
-        self.cnxn.close()
+        self._disconnect()
 
     def process_signals(self, signals):
+        host = self.host()
         output_signals = []
+        if self.cnxn is None:
+            try:
+                msg = 'Connecting to {}'.format(host)
```

Comment:
I am gathering that the "warning" here is because we aren't connected and we expected to be, I would recommend updating the log message to be consistent with that. Something like "Not connected to {host}...trying to connect now" maybe

Reply:
Good call.